src/parser.rs

```diff
@@ -234,6 +237,31 @@ fn parse_pou(
     pou
 }
 
+fn parse_generics(lexer: &mut ParseSession) -> Vec<GenericBinding> {
+    if lexer.allow(&Token::OperatorLess) {
+        parse_any_in_region(lexer, vec![Token::OperatorGreater], |lexer| {
+            let mut generics = vec![];
+            loop {
+                //identifier
+                if let Some(name) = parse_identifier(lexer) {
+                    lexer.consume_or_report(Token::KeywordColon);
+                    if let Some(nature) = parse_identifier(lexer) {
+                        generics.push(GenericBinding { name, nature });
+                    }
```

Comment:
I thinks this should be necessary rather than optional - so we need a diagnostic if the nature is missing. Should the nature's become keywords?

Reply:
I was hoping we could avoid them being extra keywords (Like we do with other types) but it might make things easier

Reply:
yes I agree, we have already a ton of keywords, but still I dont think the nature should be optional when parsing